client/webpack/generate-chunks-map-plugin.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies.
+ */
+const fs = require( 'fs' ); //eslint-disable-line import/no-nodejs-modules
+const path = require( 'path' );
+
+/**
+ * Plugin name.
+ *
+ * @type {string}
+ */
+const PLUGIN_NAME = 'GenerateChunksMap';
+
+class GenerateChunksMapPlugin {
+	constructor( { output = path.resolve( '.', 'map.json' ) } = {} ) {
+		this.output = output;
+	}
+
+	apply( compiler ) {
+		compiler.hooks.done.tap( PLUGIN_NAME, ( { compilation } ) => {
+			const { chunks } = compilation;
+
+			const chunksMap = chunks.reduce( ( map, chunk ) => {
+				const files = chunk.files || [];
+				const name = files.find( file => /\.js$/.test( file ) ) || files[ 0 ];
+
+				map[ name ] = [ ...chunk.modulesIterable ]
+					.map( ( { userRequest } ) => userRequest && path.relative( '.', userRequest ) )
```

Comment:
can we do the map / filter work prior to the assignment? I find this statement odd to read. It looks like a plain assignment that's missing a `;` at first glance.

Reply:
Fixed in https://github.com/Automattic/wp-calypso/pull/39873/commits/256ed82550d73b1e36e793120d6e48172e0517dd